numpy/ma/testutils.py

```diff
@@ -91,6 +91,21 @@ def _assert_equal_on_sequences(actual, desired, err_msg=''):
     return
 
 
+def _assert_equal_both_non_masked(actual, desired, err_msg=''):
```

Comment:
I don't think this should be a separate function, it makes the code harder to follow and is only used once.